[2025-01-24T19:56:33.679+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-24T19:56:33.692+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_clickhouse_test.spark_submit_job manual__2025-01-24T19:56:32.039504+00:00 [queued]>
[2025-01-24T19:56:33.698+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_clickhouse_test.spark_submit_job manual__2025-01-24T19:56:32.039504+00:00 [queued]>
[2025-01-24T19:56:33.699+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-01-24T19:56:33.710+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): spark_submit_job> on 2025-01-24 19:56:32.039504+00:00
[2025-01-24T19:56:33.716+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=892) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-01-24T19:56:33.717+0000] {standard_task_runner.py:72} INFO - Started process 894 to run task
[2025-01-24T19:56:33.718+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_clickhouse_test', 'spark_submit_job', 'manual__2025-01-24T19:56:32.039504+00:00', '--job-id', '108', '--raw', '--subdir', 'DAGS_FOLDER/spark_clickhouse_test.py', '--cfg-path', '/tmp/tmp7su29pqk']
[2025-01-24T19:56:33.719+0000] {standard_task_runner.py:105} INFO - Job 108: Subtask spark_submit_job
[2025-01-24T19:56:33.754+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_clickhouse_test.spark_submit_job manual__2025-01-24T19:56:32.039504+00:00 [running]> on host 39a8c5c570e1
[2025-01-24T19:56:33.812+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Smalch' AIRFLOW_CTX_DAG_ID='spark_clickhouse_test' AIRFLOW_CTX_TASK_ID='spark_submit_job' AIRFLOW_CTX_EXECUTION_DATE='2025-01-24T19:56:32.039504+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-24T19:56:32.039504+00:00'
[2025-01-24T19:56:33.814+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-24T19:56:33.821+0000] {base.py:84} INFO - Retrieving connection 'spark'
[2025-01-24T19:56:33.822+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --executor-memory 50g --driver-memory 50g --name arrow-spark --verbose --deploy-mode client dags/spark_app/spark_1.py
[2025-01-24T19:56:36.140+0000] {spark_submit.py:634} INFO - Using properties file: null
[2025-01-24T19:56:36.273+0000] {spark_submit.py:634} INFO - Parsed arguments:
[2025-01-24T19:56:36.273+0000] {spark_submit.py:634} INFO - master                  spark://spark-master:7077
[2025-01-24T19:56:36.273+0000] {spark_submit.py:634} INFO - remote                  null
[2025-01-24T19:56:36.274+0000] {spark_submit.py:634} INFO - deployMode              client
[2025-01-24T19:56:36.274+0000] {spark_submit.py:634} INFO - executorMemory          50g
[2025-01-24T19:56:36.274+0000] {spark_submit.py:634} INFO - executorCores           null
[2025-01-24T19:56:36.274+0000] {spark_submit.py:634} INFO - totalExecutorCores      null
[2025-01-24T19:56:36.274+0000] {spark_submit.py:634} INFO - propertiesFile          null
[2025-01-24T19:56:36.275+0000] {spark_submit.py:634} INFO - driverMemory            50g
[2025-01-24T19:56:36.275+0000] {spark_submit.py:634} INFO - driverCores             null
[2025-01-24T19:56:36.275+0000] {spark_submit.py:634} INFO - driverExtraClassPath    null
[2025-01-24T19:56:36.275+0000] {spark_submit.py:634} INFO - driverExtraLibraryPath  null
[2025-01-24T19:56:36.276+0000] {spark_submit.py:634} INFO - driverExtraJavaOptions  null
[2025-01-24T19:56:36.276+0000] {spark_submit.py:634} INFO - supervise               false
[2025-01-24T19:56:36.276+0000] {spark_submit.py:634} INFO - queue                   null
[2025-01-24T19:56:36.276+0000] {spark_submit.py:634} INFO - numExecutors            null
[2025-01-24T19:56:36.276+0000] {spark_submit.py:634} INFO - files                   null
[2025-01-24T19:56:36.277+0000] {spark_submit.py:634} INFO - pyFiles                 null
[2025-01-24T19:56:36.277+0000] {spark_submit.py:634} INFO - archives                null
[2025-01-24T19:56:36.277+0000] {spark_submit.py:634} INFO - mainClass               null
[2025-01-24T19:56:36.277+0000] {spark_submit.py:634} INFO - primaryResource         file:/opt/***/dags/spark_app/spark_1.py
[2025-01-24T19:56:36.277+0000] {spark_submit.py:634} INFO - name                    arrow-spark
[2025-01-24T19:56:36.278+0000] {spark_submit.py:634} INFO - childArgs               []
[2025-01-24T19:56:36.278+0000] {spark_submit.py:634} INFO - jars                    null
[2025-01-24T19:56:36.278+0000] {spark_submit.py:634} INFO - packages                null
[2025-01-24T19:56:36.278+0000] {spark_submit.py:634} INFO - packagesExclusions      null
[2025-01-24T19:56:36.278+0000] {spark_submit.py:634} INFO - repositories            null
[2025-01-24T19:56:36.279+0000] {spark_submit.py:634} INFO - verbose                 true
[2025-01-24T19:56:36.279+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:36.279+0000] {spark_submit.py:634} INFO - Spark properties used, including those specified through
[2025-01-24T19:56:36.279+0000] {spark_submit.py:634} INFO - --conf and those from the properties file null:
[2025-01-24T19:56:36.280+0000] {spark_submit.py:634} INFO - (spark.driver.memory,50g)
[2025-01-24T19:56:36.280+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:36.280+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:36.769+0000] {spark_submit.py:634} INFO - Main class:
[2025-01-24T19:56:36.770+0000] {spark_submit.py:634} INFO - org.apache.spark.deploy.PythonRunner
[2025-01-24T19:56:36.770+0000] {spark_submit.py:634} INFO - Arguments:
[2025-01-24T19:56:36.771+0000] {spark_submit.py:634} INFO - file:/opt/***/dags/spark_app/spark_1.py
[2025-01-24T19:56:36.771+0000] {spark_submit.py:634} INFO - null
[2025-01-24T19:56:36.774+0000] {spark_submit.py:634} INFO - Spark config:
[2025-01-24T19:56:36.775+0000] {spark_submit.py:634} INFO - (spark.app.name,arrow-spark)
[2025-01-24T19:56:36.775+0000] {spark_submit.py:634} INFO - (spark.app.submitTime,1737748596752)
[2025-01-24T19:56:36.775+0000] {spark_submit.py:634} INFO - (spark.driver.memory,50g)
[2025-01-24T19:56:36.775+0000] {spark_submit.py:634} INFO - (spark.executor.memory,50g)
[2025-01-24T19:56:36.775+0000] {spark_submit.py:634} INFO - (spark.master,spark://spark-master:7077)
[2025-01-24T19:56:36.776+0000] {spark_submit.py:634} INFO - (spark.submit.deployMode,client)
[2025-01-24T19:56:36.776+0000] {spark_submit.py:634} INFO - (spark.submit.pyFiles,)
[2025-01-24T19:56:36.788+0000] {spark_submit.py:634} INFO - Classpath elements:
[2025-01-24T19:56:36.789+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:36.790+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:36.791+0000] {spark_submit.py:634} INFO - 
[2025-01-24T19:56:38.144+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkContext: Running Spark version 3.5.4
[2025-01-24T19:56:38.145+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkContext: OS info Linux, 5.10.0-33-cloud-amd64, amd64
[2025-01-24T19:56:38.146+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkContext: Java version 17.0.13
[2025-01-24T19:56:38.237+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-01-24T19:56:38.359+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceUtils: ==============================================================
[2025-01-24T19:56:38.359+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-01-24T19:56:38.360+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceUtils: ==============================================================
[2025-01-24T19:56:38.360+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkContext: Submitted application: ExampleJob111
[2025-01-24T19:56:38.383+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 51200, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-01-24T19:56:38.391+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceProfile: Limiting resource is cpu
[2025-01-24T19:56:38.392+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-01-24T19:56:38.475+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SecurityManager: Changing view acls to: default
[2025-01-24T19:56:38.475+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SecurityManager: Changing modify acls to: default
[2025-01-24T19:56:38.476+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SecurityManager: Changing view acls groups to:
[2025-01-24T19:56:38.476+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SecurityManager: Changing modify acls groups to:
[2025-01-24T19:56:38.477+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: default; groups with view permissions: EMPTY; users with modify permissions: default; groups with modify permissions: EMPTY
[2025-01-24T19:56:38.780+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO Utils: Successfully started service 'sparkDriver' on port 39917.
[2025-01-24T19:56:38.824+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkEnv: Registering MapOutputTracker
[2025-01-24T19:56:38.871+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkEnv: Registering BlockManagerMaster
[2025-01-24T19:56:38.895+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-01-24T19:56:38.896+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-01-24T19:56:38.901+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-01-24T19:56:38.927+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-9d2cebbb-5716-4cee-9eaf-29b3e72bce16
[2025-01-24T19:56:38.941+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO MemoryStore: MemoryStore started with capacity 29.8 GiB
[2025-01-24T19:56:38.954+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:38 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-01-24T19:56:39.116+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-01-24T19:56:39.170+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-01-24T19:56:39.335+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-01-24T19:56:39.408+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.8:7077 after 39 ms (0 ms spent in bootstraps)
[2025-01-24T19:56:39.545+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250124195639-0000
[2025-01-24T19:56:39.555+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 35695.
[2025-01-24T19:56:39.556+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO NettyBlockTransferService: Server created on 39a8c5c570e1:35695
[2025-01-24T19:56:39.557+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-01-24T19:56:39.563+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 39a8c5c570e1, 35695, None)
[2025-01-24T19:56:39.565+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO BlockManagerMasterEndpoint: Registering block manager 39a8c5c570e1:35695 with 29.8 GiB RAM, BlockManagerId(driver, 39a8c5c570e1, 35695, None)
[2025-01-24T19:56:39.567+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 39a8c5c570e1, 35695, None)
[2025-01-24T19:56:39.568+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 39a8c5c570e1, 35695, None)
[2025-01-24T19:56:39.770+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:39 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-01-24T19:56:40.333+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:40 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-01-24T19:56:40.339+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:40 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-01-24T19:56:42.489+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO ParquetUtils: Using default output committer for Parquet: org.apache.parquet.hadoop.ParquetOutputCommitter
[2025-01-24T19:56:42.509+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO FileOutputCommitter: File Output Committer Algorithm version is 1
[2025-01-24T19:56:42.509+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO FileOutputCommitter: FileOutputCommitter skip cleanup _temporary folders under output directory:false, ignore cleanup failures: false
[2025-01-24T19:56:42.510+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO SQLHadoopMapReduceCommitProtocol: Using user defined output committer class org.apache.parquet.hadoop.ParquetOutputCommitter
[2025-01-24T19:56:42.510+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO FileOutputCommitter: File Output Committer Algorithm version is 1
[2025-01-24T19:56:42.510+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO FileOutputCommitter: FileOutputCommitter skip cleanup _temporary folders under output directory:false, ignore cleanup failures: false
[2025-01-24T19:56:42.511+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO SQLHadoopMapReduceCommitProtocol: Using output committer class org.apache.parquet.hadoop.ParquetOutputCommitter
[2025-01-24T19:56:42.746+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO CodeGenerator: Code generated in 180.60562 ms
[2025-01-24T19:56:42.773+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO SparkContext: Starting job: parquet at NativeMethodAccessorImpl.java:0
[2025-01-24T19:56:42.784+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Got job 0 (parquet at NativeMethodAccessorImpl.java:0) with 2 output partitions
[2025-01-24T19:56:42.784+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Final stage: ResultStage 0 (parquet at NativeMethodAccessorImpl.java:0)
[2025-01-24T19:56:42.784+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Parents of final stage: List()
[2025-01-24T19:56:42.786+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Missing parents: List()
[2025-01-24T19:56:42.788+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[6] at parquet at NativeMethodAccessorImpl.java:0), which has no missing parents
[2025-01-24T19:56:42.892+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 213.8 KiB, free 29.8 GiB)
[2025-01-24T19:56:42.917+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 77.7 KiB, free 29.8 GiB)
[2025-01-24T19:56:42.919+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 39a8c5c570e1:35695 (size: 77.7 KiB, free: 29.8 GiB)
[2025-01-24T19:56:42.921+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:1585
[2025-01-24T19:56:42.934+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO DAGScheduler: Submitting 2 missing tasks from ResultStage 0 (MapPartitionsRDD[6] at parquet at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0, 1))
[2025-01-24T19:56:42.935+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:42 INFO TaskSchedulerImpl: Adding task set 0.0 with 2 tasks resource profile 0
[2025-01-24T19:56:57.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:56:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:57:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:57:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:57:27.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:57:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:57:42.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:57:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:57:57.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:57:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:58:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:58:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:58:27.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:58:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:58:42.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:58:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:58:57.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:58:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:59:12.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:59:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:59:27.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:59:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:59:42.943+0000] {spark_submit.py:634} INFO - 25/01/24 19:59:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T19:59:57.942+0000] {spark_submit.py:634} INFO - 25/01/24 19:59:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:00:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:00:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:00:27.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:00:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:00:42.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:00:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:00:57.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:00:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:01:12.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:01:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:01:27.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:01:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:01:42.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:01:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:01:57.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:01:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:02:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:02:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:02:27.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:02:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:02:42.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:02:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:02:57.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:02:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:03:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:03:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:03:27.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:03:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:03:42.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:03:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:03:57.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:03:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:04:12.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:04:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:04:27.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:04:27 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:04:42.943+0000] {spark_submit.py:634} INFO - 25/01/24 20:04:42 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:04:57.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:04:57 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:05:12.942+0000] {spark_submit.py:634} INFO - 25/01/24 20:05:12 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-24T20:05:17.634+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to None. Terminating instance.
[2025-01-24T20:05:17.634+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-01-24T20:05:17.635+0000] {process_utils.py:132} INFO - Sending 15 to group 894. PIDs of all processes in the group: [896, 956, 894]
[2025-01-24T20:05:17.635+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 894
[2025-01-24T20:05:17.635+0000] {taskinstance.py:3093} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-01-24T20:05:17.636+0000] {spark_submit.py:760} INFO - Sending kill signal to spark-submit
[2025-01-24T20:05:17.646+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 548, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 597, in _process_spark_submit_log
    for line in itr:
                ^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3095, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2025-01-24T20:05:17.651+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=spark_clickhouse_test, task_id=spark_submit_job, run_id=manual__2025-01-24T19:56:32.039504+00:00, execution_date=20250124T195632, start_date=20250124T195633, end_date=20250124T200517
[2025-01-24T20:05:17.650+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=956, status='terminated', started='19:56:36') (956) terminated with exit code None
[2025-01-24T20:05:17.669+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-24T20:05:17.670+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 108 for task spark_submit_job ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(spark_clickhouse_test, spark_submit_job, manual__2025-01-24T19:56:32.039504+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'spark_submit_job', 'dag_id': 'spark_clickhouse_test', 'run_id': 'manual__2025-01-24T19:56:32.039504+00:00', 'map_index': -1, 'start_date': datetime.datetime(2025, 1, 24, 19, 56, 33, 692756, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2025, 1, 24, 20, 5, 17, 651229, tzinfo=Timezone('UTC')), 'duration': 523}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 894)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 548, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 597, in _process_spark_submit_log
    for line in itr:
                ^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3095, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.ForeignKeyViolation: insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(spark_clickhouse_test, spark_submit_job, manual__2025-01-24T19:56:32.039504+00:00, -1) is not present in table "task_instance".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 327, in _run_raw_task
    ti.handle_failure(e, test_mode, context, session=session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3428, in handle_failure
    _handle_failure(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 1243, in _handle_failure
    TaskInstance.save_to_db(failure_context["ti"], session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3400, in save_to_db
    session.flush()
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 3449, in flush
    self._flush(objects)
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 3588, in _flush
    with util.safe_reraise():
         ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 3549, in _flush
    flush_context.execute()
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
    rec.execute(self)
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    _emit_insert_statements(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/persistence.py", line 1238, in _emit_insert_statements
    result = connection._execute_20(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1710, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(spark_clickhouse_test, spark_submit_job, manual__2025-01-24T19:56:32.039504+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'spark_submit_job', 'dag_id': 'spark_clickhouse_test', 'run_id': 'manual__2025-01-24T19:56:32.039504+00:00', 'map_index': -1, 'start_date': datetime.datetime(2025, 1, 24, 19, 56, 33, 692756, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2025, 1, 24, 20, 5, 17, 651229, tzinfo=Timezone('UTC')), 'duration': 523}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-01-24T20:05:17.704+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=896, status='terminated', started='19:56:33') (896) terminated with exit code None
[2025-01-24T20:05:17.705+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=894, status='terminated', exitcode=1, started='19:56:33') (894) terminated with exit code 1
